src/Drush/Generators/AdminReadmeGridGenerator.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\tripal_devtools\Drush\Generators;
+
+use DrupalCodeGenerator\Asset\AssetCollection as Assets;
+use DrupalCodeGenerator\Attribute\Generator;
+use DrupalCodeGenerator\Command\BaseGenerator;
+use DrupalCodeGenerator\GeneratorType;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Generate README grid.
+ */
+#[Generator(
+  name: 'tripal-admin:readme-grid',
+  description: 'Generates the Drupal by PHP version compatibility grid for the README and associated GitHub workflow files.',
+  templatePath: __DIR__ . '/../../../templates/generator/tripal_admin',
+  type: GeneratorType::MODULE_COMPONENT,
+)]
+final class AdminReadmeGridGenerator extends BaseGenerator {
+
+  /**
+   * The main workflow file.
+   *
+   * @var string
+   */
+  private const WORKFLOW_FILE = 'ALL-phpunit.yml';
+
+  /**
+   * The main workflow directory.
+   *
+   * @var string
+   */
+  private const WORKFLOW_DIR = '.github/workflows';
+
+  /**
+   * Workflow stack version YML keys.
+   *
+   * @var array
+   */
+  private const WORKFLOW_VERSION = [
+    'php' => 'php-version',
+    'drupal' => 'drupal-version',
+    'pgsql' => 'pgsql-version',
+  ];
+
+  /**
+   * Workflow configuration values.
+   *
+   * Each key corresponds to a config key in the template file.
+   *
+   * @see /templates/tripal_admin/readme-grid-workflow.twig
+   *
+   * @var string
+   */
+  private const WORKFLOW_OPTION = [
+    'name' => 'PHPUnit',
+    'branches' => [
+      '4.x',
+    ],
+    'cron' => '0 6 * * *',
+    'test' => 'running-tests',
+    'checkout' => 'actions/checkout@v4',
+    'run' => 'tripal/test-tripal-action@v1.7',
+  ];
+
+  /**
+   * The workflow template file.
+   *
+   * @var string
+   */
+  private const WORKFLOW_TEMPLATE = 'readme-grid-workflow.twig';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generate(array &$vars, Assets $assets): void {
+
+    $ir = $this->createInterviewer($vars);
+    $machine_name = $ir->askMachineName();
+
+    // Must have this key.
+    $vars['machine_name'] = $machine_name;
+
+    $module = \Drupal::service('module_handler')
+      ->getModule($machine_name);
+    if (!$module) {
+      throw new \Exception('Module does not exist.');
+    }
+
+    $phpunit_yml = '%s' . DIRECTORY_SEPARATOR . self::WORKFLOW_DIR . DIRECTORY_SEPARATOR . self::WORKFLOW_FILE;
```

Comment:
It might be good to let the user/developer know the assumption we are making here. The following suggestion provides the output shown in the screenshot.


```suggestion
    $phpunit_yml = self::WORKFLOW_DIR . DIRECTORY_SEPARATOR . self::WORKFLOW_FILE;
    $this->io()->note("We use the PHPUnit matrixed workflow to determine the combinations being tested.");
    $this->io()->note("Specifically, we expect there to be a Github Workflow ($phpunit_yml) with the 'jobs.run-tests.strategy.matrix' defined.");
    $phpunit_yml = '%s' . DIRECTORY_SEPARATOR . $phpunit_yml;
```

<img width="1040" height="215" alt="Image" src="https://github.com/user-attachments/assets/8c8d5caa-2867-4d92-bc57-4a01e2a8fe97" />